### PR TITLE
reset_password

### PR DESCRIPTION
**Pull request by weilingooi**
done reset_password

### CODE REVIEW

**Comment by weilingooi** [comment]
ready

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,3 +1,26 @@
 class ApplicationController < ActionController::Base
   include SessionsHelper
+  
+  before_action :switch_locale
```
Comment (chungpv-1008) [inline app/controllers/application_controller.rb:4]:
```suggestion
  before_action :switch_locale

```

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,3 +1,26 @@
...
+  def default_url_options
+    {locale: I18n.locale}
+  end
+
+  private
+  def switch_locale
```
Comment (chungpv-1008) [inline app/controllers/application_controller.rb]:
```suggestion
  private

  def switch_locale
```

File: app/controllers/microposts_controller.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    end
+    redirect_to request.referer || root_path
+  end
+
+  private
+
+  def correct_user
+    @micropost = current_user.microposts.find_by id: params[:id]
+    return if @micropost
+    flash[:danger] = t "sessions.incorrect_user"
```
Comment (chungpv-1008) [inline app/controllers/microposts_controller.rb]:
```suggestion
    return if @micropost
 
    flash[:danger] = t "sessions.incorrect_user"
```

File: app/controllers/microposts_controller.rb [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    redirect_to root_path
+  end
+
+  def check_login
+    return if logged_in
+    store_location
```
Comment (chungpv-1008) [inline app/controllers/microposts_controller.rb]: add blank line after 'return'

File: app/controllers/microposts_controller.rb [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  before_action :correct_user, only: :destroy
+
+  def create
+    @micropost = current_user.microposts.build create_micropost_params
+    @micropost.image.attach create_micropost_params[:image]
+    if @micropost.save
+      flash[:success] = t "microposts.create_success"
+      redirect_to root_path
+    else
+      flash[:warning] = t "microposts.create_fail"
```
Comment (chungpv-1008) [inline app/controllers/microposts_controller.rb]:
```suggestion
      flash.now[:warning] = t "microposts.create_fail"
```

File: app/controllers/static_pages_controller.rb [hunk 6]
```diff
@@ -1,5 +1,10 @@
 class StaticPagesController < ApplicationController
-  def home; end
+  def home
+  	return unless logged_in?
```
Comment (chungpv-1008) [inline app/controllers/static_pages_controller.rb]: indent

File: app/javascript/packs/uploadImage.js [hunk 7]
```diff
@@ -0,0 +1,8 @@
+$(function () {
```
Comment (chungpv-1008) [inline app/javascript/packs/uploadImage.js:1]: add end line for this file

File: app/models/micropost.rb [hunk 8]
```diff
@@ -0,0 +1,12 @@
+class Micropost < ApplicationRecord
+  belongs_to :user
+  has_one_attached :image
+  validates :user_id, presence: true
+  validates :content, presence: true, length: {maximum: Settings.validations.content.max_length}
+  validates :image, content_type: { in: Settings.files.image_type, message: I18n.t("errors.messages.invalid_pic") }, size: { less_than: Settings.files.pic_size.megabytes, message: I18n.t("errors.messages.less_than") }
+  scope :recent_posts, ->{order created_at: :desc}
```
Comment (chungpv-1008) [inline app/models/micropost.rb:7]: please arrange follow
Screenshot from 2020-08-26 08-11-15

File: app/models/micropost.rb [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+
+  def display_image
+  image.variant resize_to_limit: [Settings.files.pic_resize, Settings.files.pic_resize]
```
Comment (chungpv-1008) [inline app/models/micropost.rb]:
```suggestion
    image.variant resize_to_limit: [Settings.files.pic_resize, Settings.files.pic_resize]
```

File: app/views/shared/_user_info.html.erb [hunk 10]
```diff
@@ -0,0 +1,4 @@
+<%= link_to gravatar_for(current_user), current_user %>
+<h1><%= current_user.name %></h1>
+<span><%= link_to t(".profile"), current_user %></span>
+<span><%= pluralize current_user.micro_posts.count, t("microposts.title") %></span>
```
Comment (chungpv-1008) [inline app/views/shared/_user_info.html.erb]:
```suggestion
<span><%= pluralize current_user.micro_posts.size, t("microposts.title") %></span>
```

**Comment by weilingooi** [comment]
ready

**Review by chungpv-1008 — COMMENTED** [review]
merged

**Pull request merged**